db/migrations/20200603124229_create_time_ilk_snapshot_query.sql

```diff
@@ -0,0 +1,66 @@
+-- +goose Up
+CREATE TYPE api.time_ilk_snapshot AS
+(
+    bucket_start TIMESTAMP,
+    bucket_end TIMESTAMP,
+    bucket_interval INTERVAL,
+    ilk_identifier TEXT,
+    block_number BIGINT,
+    rate NUMERIC,
+    art NUMERIC,
+    spot NUMERIC,
+    line NUMERIC,
+    dust NUMERIC,
+    chop NUMERIC,
+    lump NUMERIC,
+    flip TEXT,
+    rho NUMERIC,
+    duty NUMERIC,
+    pip TEXT,
+    mat NUMERIC,
+    created TIMESTAMP,
+    updated TIMESTAMP
+);
+
+CREATE FUNCTION api.time_ilk_snapshots(ilk_identifier TEXT, range_start TIMESTAMP, range_end TIMESTAMP, bucket_interval INTERVAL DEFAULT '1 day'::INTERVAL)
+    RETURNS SETOF api.time_ilk_snapshot AS
+$$
+WITH buckets AS (SELECT generate_series(range_start, range_end - bucket_interval, bucket_interval) AS bucket_start)
+SELECT buckets.bucket_start AS bucket_start,
+       buckets.bucket_start + bucket_interval AS bucket_end,
+       bucket_interval AS bucket_interval,
```

Comment:
Why do we need the `AS` here?

Reply:
I was adding it for explicit consistency, but I can alter the style to omit when it's not necessary.